src/app/components/user-list/user-list.component.ts

```diff
@@ -8,33 +9,34 @@ import { WebSocketService } from 'src/app/services/web-socket.service';
 })
 export class UserListComponent implements OnInit{
   
-  users : Record<string,string> = {};
+  users : Array<UserInfo> = [];
 
   constructor(private webSocketService : WebSocketService){}
 
   ngOnInit(): void {
     this.webSocketService.userList$.subscribe((user)=>{
-      this.users = {...user};
+      this.users=[];
+      this.users = [...user];
     })
   }
 
-  changeReceiver(receiver : string){
+  changeReceiver(receiver : UserInfo){
     this.webSocketService.changeReceiver(receiver);
   }
 
-  isCurrentUser(user:string){
-    return this.webSocketService.username === user;
+  isCurrentUser(user:UserInfo){    
+    return this.webSocketService.userid === user.id;
   }
 
-  isCurrentReceiver(user : string){
-    return this.webSocketService.receiverUsername === user;
+  isCurrentReceiver(user : UserInfo){
+    return this.webSocketService.receiverUserid === user.id;
   }
 
   get userStatuses() {
```

Comment:
Remove this property getter as its not needed

Reply:
Done